src/hooks/usePermit2.ts

```diff
@@ -0,0 +1,85 @@
+import { PERMIT2_ADDRESS } from '@uniswap/permit2-sdk'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { useWeb3React } from '@web3-react/core'
+import ms from 'ms.macro'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { ApprovalTransactionInfo } from '..'
+import useInterval from './useInterval'
+import { PermitSignature, usePermitAllowance, useUpdatePermitAllowance } from './usePermitAllowance'
+import { useTokenAllowance, useUpdateTokenAllowance } from './useTokenAllowance'
+
+export enum PermitState {
+  UNKNOWN,
+  APPROVAL_NEEDED,
+  PERMIT_NEEDED,
+  PERMITTED,
+}
+
+export interface Permit {
+  state: PermitState
+  signature?: PermitSignature
+  callback?: () => Promise<ApprovalTransactionInfo | void>
+}
+
+export default function usePermit(amount?: CurrencyAmount<Token>, spender?: string): Permit {
+  const { account } = useWeb3React()
+  const tokenAllowance = useTokenAllowance(amount?.currency, account, PERMIT2_ADDRESS)
+  const updateTokenAllowance = useUpdateTokenAllowance(amount, PERMIT2_ADDRESS)
+
+  const allowanceData = usePermitAllowance(amount?.currency, spender)
+  const [permitAllowance, setPermitAllowance] = useState(allowanceData?.amount)
+  useEffect(() => setPermitAllowance(allowanceData?.amount), [allowanceData?.amount])
+
+  const [signature, setSignature] = useState<PermitSignature>()
+  const updatePermitAllowance = useUpdatePermitAllowance(amount?.currency, spender, allowanceData?.nonce, setSignature)
+
+  const updateTokenAndPermitAllowance = useCallback(async () => {
+    // Queue both transactions.
```

Comment:
this comment is confusing because it implies that the next line queues both transactions